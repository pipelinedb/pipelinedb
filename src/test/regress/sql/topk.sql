SELECT topk(topk_agg(x, 10)) FROM generate_series(1, 100) AS x;
SELECT topk_values(topk_agg(x, 10)) FROM ( 
  SELECT generate_series(1, 100)
    UNION ALL
  SELECT generate_series(1, 10)
    UNION ALL
  SELECT generate_series(1, 30)) AS x;

SELECT topk_freqs(topk_agg(x, 10)) FROM ( 
  SELECT generate_series(1, 100)
    UNION ALL
  SELECT generate_series(1, 10)
    UNION ALL
  SELECT generate_series(1, 30)) AS x;

SELECT topk(topk_agg(x, 10)) FROM ( 
  SELECT generate_series(1, 100)
    UNION ALL
  SELECT generate_series(1, 10)
    UNION ALL
  SELECT generate_series(1, 30)) AS x;
