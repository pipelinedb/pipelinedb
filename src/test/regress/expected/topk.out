SELECT topk(topk_agg(x, 10)) FROM generate_series(1, 100) AS x;
  topk  
--------
 (40,2)
 (45,2)
 (47,2)
 (56,2)
 (58,2)
 (79,2)
 (80,2)
 (82,2)
 (83,2)
 (86,2)
(10 rows)

SELECT topk_values(topk_agg(x, 10)) FROM ( 
  SELECT generate_series(1, 100)
    UNION ALL
  SELECT generate_series(1, 10)
    UNION ALL
  SELECT generate_series(1, 30)) AS x;
                topk_values                 
--------------------------------------------
 {(1),(4),(8),(5),(6),(3),(2),(7),(9),(10)}
(1 row)

SELECT topk_freqs(topk_agg(x, 10)) FROM ( 
  SELECT generate_series(1, 100)
    UNION ALL
  SELECT generate_series(1, 10)
    UNION ALL
  SELECT generate_series(1, 30)) AS x;
      topk_freqs       
-----------------------
 {3,3,3,3,3,3,3,3,3,3}
(1 row)

SELECT topk(topk_agg(x, 10)) FROM ( 
  SELECT generate_series(1, 100)
    UNION ALL
  SELECT generate_series(1, 10)
    UNION ALL
  SELECT generate_series(1, 30)) AS x;
    topk    
------------
 ("(1)",3)
 ("(4)",3)
 ("(8)",3)
 ("(5)",3)
 ("(6)",3)
 ("(3)",3)
 ("(2)",3)
 ("(7)",3)
 ("(9)",3)
 ("(10)",3)
(10 rows)

